tests/secret-providers/azure_keyvault_workload_identity/azure_keyvault_workload_identity_test.go

```diff
@@ -0,0 +1,235 @@
+//go:build e2e
+// +build e2e
+
+package azure_keyvault_workload_identity_test
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-storage-queue-go/azqueue"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/scalers/azure"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "azure-keyvault-workload-identity-queue-test"
+)
+
+var (
+	connectionString = os.Getenv("AZURE_STORAGE_CONNECTION_STRING")
+	keyvaultURI      = os.Getenv("AZURE_KEYVAULT_URI")
+	azureADClientID  = os.Getenv("AZURE_SP_APP_ID")
+	azureADTenantID  = os.Getenv("AZURE_SP_TENANT")
+	testNamespace    = fmt.Sprintf("%s-ns", testName)
+	secretName       = fmt.Sprintf("%s-secret", testName)
+	deploymentName   = fmt.Sprintf("%s-deployment", testName)
+	triggerAuthName  = fmt.Sprintf("%s-ta", testName)
+	scaledObjectName = fmt.Sprintf("%s-so", testName)
+	queueName        = fmt.Sprintf("%s-queue-%d", testName, GetRandomNumber())
+)
+
+type templateData struct {
+	TestNamespace    string
+	SecretName       string
+	Connection       string
+	DeploymentName   string
+	TriggerAuthName  string
+	ScaledObjectName string
+	QueueName        string
+	KeyVaultURI      string
+	AzureADClientID  string
+	AzureADTenantID  string
```

Comment:
Delete unused fields.

Reply:
Lol, this week is not my week... Thanks for catching these things

Reply:
Least I can do for a bug I created :D

Reply:
Don't worry at all, these things happen from time to time, as I said, I broke (literaly) half RabbitMQ scaler on my first contribution (it was also my second PR week with golang xD)